common/utilities/timing.py

```diff
@@ -0,0 +1,127 @@
+import logging
+import time
+from functools import wraps
+
+
+class StopWatch:
+
+    start_time = None
+
+    def start_timer(self):
+        self.start_time = time.time()
+
+    def stop_timer(self):
+        end_time = time.time()
+        diff = end_time - self.start_time
+        return diff
+
+
+def time_method(method):
+    """
+    A method decorator that logs the time taken to execute a given method
+
+    :param method:
+    :return:
+    """
+
+    @wraps(method)
+    def invoke_method_with_timer(*args, **kwargs):
+        stopwatch = StopWatch()
+        stopwatch.start_timer()
+        try:
+            return method(*args, **kwargs)
+        finally:
+            duration = stopwatch.stop_timer()
+            duration = round(duration, 3)
+            logging.info({
```

Comment:
Just looking at this again, this will produce output like:

```
2019-07-16 10:38:40,836 - INFO: {'method': 'foo', 'duration': 12.345}
```

This doesn't tally with our [logging guidance](https://gpitbjss.atlassian.net/wiki/spaces/RTDel/pages/1560608910/KDD-001+-+Logging+Strategy+and+Format) around log line format (in the Log Line Format section):

> Log lines should be human readable strings, not blobs of JSON or XML

This should log something more like `Method name=foo took duration=12.345 seconds`.

Reply:
This is correct, but that is something that is expected to be handled as apart of the logging module that is to be written next. I think when we started this ticket it was under the expectation that the logging would change/be updated following the proper logging implementation

Reply:
I'd suggest that it's probably better to write the log message in the expected output format now, rather than having to come back and re-write it when the logging module is plugged in.